### DISCUSSION TITLE
I don't understand something about the notes of InstanceSymbol::resolvePortConnections

### DISCUSSION THREAD

**Discussion by luckyhorse1** [MikePopoloski/slang#374] [Q&A] [answered]
## the notes of InstanceSymbol::resolvePortConnections
```c++
 //     interface I #(parameter int i) (); endinterface
  //     module M(I iface, input logic [iface.i - 1 : 0] foo);
  //         localparam int j = $bits(foo);
  //     endmodule
  //
  // In order to resolve connections for an instance of M, we elaborate its body,
  // which then requires evaluating $bits(foo) which then depends on the connection
  // provided to `iface`.
```
- why  resolve connections need to elaborate its body?
- for ansi-port，just like above，there is no need to elaborate body when resolve connections.
- for non-ansi-port, there is no interface port. so, this does not exist

**Comment by MikePopoloski** [comment]
Non-ansi interface ports are allowed in SystemVerilog. I haven't added support for them yet but I will at some point.

**Reply by luckyhorse1** [reply]
Thank you for your reply.   
In fact, I debugged this example.
```c++
//     interface I #(parameter int i) (); endinterface
//     module M(I iface, input logic [iface.i - 1 : 0] foo);
//         localparam int j = $bits(foo);
//     endmodule
```
But, when resolve connections for an instance of M, just handle port list. Not handle foo‘s type  logic [iface.i - 1 : 0] and j's initializer $bits.    
Now, when resolve connections finished, will visit M's body and M's member.  
Then, foo‘s type  logic [iface.i - 1 : 0] is handled when handle port symbol.   
 j's initializer $bits is handled when handle parameter symbol.
So, I cannot understand  you say that "In the code, this translates to a reetrant call to this  function; the first time we call getPortList() on the body will call back in here".

**Comment by MikePopoloski** [comment, marked as answer]
I'm not sure what to tell you; the function is in fact reentrant when making port connections. The full test example that shows this is:
```
interface I #(parameter int i) ();
endinterface

module M(I iface, input logic [iface.i - 1 : 0] foo);
    localparam int j = $bits(foo);
endmodule

module test;
    I #(17) i();
    M m(i, 1);
endmodule
```
Setting a break point in resolvePortConnections will show you the call stack where this occurs.

**Reply by luckyhorse1** [reply]
You are wright ,thank you!
